### DISCUSSION TITLE
Firebase Keys and the .split method

### DISCUSSION THREAD

**Discussion by caseywalker** [nss-evening-cohort-14/evening-client-side#157] [Look at what I learned!]
I implemented a delete button on the Pinterest project, however - three of my pins just would not delete. All of my others would, except 3 pesky pins. After a period of trouble shooting, I came to find that my firebaseKeys includes '--' within the key.. which was the same string I was splitting on delete. 

Example. firebaseKey: "-MV--ro-YXXQFivU6n5j"
    ``` <a href="#" class="btn btn-danger" id="deletePin--${item.firebaseKey}--${item.board_id}">Delete Pin</a>```
const firebaseKey = e.target.id.split('--')[1] = "-MV" 

Resolution: 
```<a href="#" class="btn btn-danger" id="deletePin^^${item.firebaseKey}^^${item.board_id}">Delete Pin</a>```
const firebaseKey = e.target.id.split('^^')[1]

**Comment by drteresavasquez** [comment]
That's awesome. Were those keys generated by Firebase? They usually will not do that on their keys which is why I asked. 

Great work around. I love to see it! 🔥🔥

**Reply by caseywalker** [reply]
Yep, I was using Postman to post my sample data, and patched the response as the firebaseKey. 

3 of my sample pins I posted back to back all had the same issue. I wonder if there is a sequence or timestamp that generated that.

**Reply by drteresavasquez** [reply]
That’s interesting but knowing that you can split on anything means you’re understanding!!! I’m so excited about that!  It happens that’s for sure!
